k9mail/src/main/java/com/fsck/k9/activity/misc/ContactPictureLoader.java

```diff
@@ -110,59 +111,70 @@ public ContactPictureLoader(Context context, int defaultBackgroundColor) {
-     *
-     * <p>
-     * If a picture is found in the cache, it is displayed in the {@code ContactBadge}
-     * immediately. Otherwise a {@link ContactPictureRetrievalTask} is started to try to load the
-     * contact picture in a background thread. Depending on the result the contact picture or a
-     * fallback picture is then stored in the bitmap cache.
-     * </p>
-     *
-     * @param address
-     *         The {@link Address} instance holding the email address that is used to search the
-     *         contacts database.
-     * @param imageView
-     *         The {@code ContactBadge} instance to receive the picture.
-     *
-     * @see #mBitmapCache
-     * @see #calculateFallbackBitmap(Address)
-     */
-    public void loadContactPicture(Address address, ImageView imageView) {
-        Bitmap bitmap = getBitmapFromCache(address);
-        if (bitmap != null) {
-            // The picture was found in the bitmap cache
-            imageView.setImageBitmap(bitmap);
-        } else if (cancelPotentialWork(address, imageView)) {
-            // Query the contacts database in a background thread and try to load the contact
-            // picture, if there is one.
-            ContactPictureRetrievalTask task = new ContactPictureRetrievalTask(imageView, address);
-            AsyncDrawable asyncDrawable = new AsyncDrawable(mResources,
-                    calculateFallbackBitmap(address), task);
-            imageView.setImageDrawable(asyncDrawable);
-            try {
-                task.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
-            } catch (RejectedExecutionException e) {
-                // We flooded the thread pool queue... use a fallback picture
-                imageView.setImageBitmap(calculateFallbackBitmap(address));
-            }
+    private void loadFallbackPicture(Address address, ImageView imageView) {
+        Context context = imageView.getContext();
+
+        Glide.with(context)
+                .using(new FallbackGlideModelLoader(), FallbackGlideParams.class)
+                .from(FallbackGlideParams.class)
+                .as(Bitmap.class)
+                .transcode(new BitmapToGlideDrawableTranscoder(context), GlideDrawable.class)
+                .decoder(new FallbackGlideBitmapDecoder(context))
+                .encoder(new BitmapEncoder(Bitmap.CompressFormat.PNG, 0))
+                .cacheDecoder(new FileToStreamDecoder<>(new StreamBitmapDecoder(context)))
+                .diskCacheStrategy(DiskCacheStrategy.NONE)
+                .load(new FallbackGlideParams(address))
+                // for some reason, following 2 lines fix loading issues.
+                .dontAnimate()
+                .override(mPictureSizeInPx, mPictureSizeInPx)
+                .into(imageView);
+    }
+
+    public void loadContactPicture(final Recipient recipient, final ImageView imageView) {
+        if (recipient.photoThumbnailUri != null) {
```

Comment:
this method seems to do roughly the same thing as loadContactPicture called with an Address argument, instead of Recipient. wouldn't it be easier to just call that method here? or is there an important semantic difference between the two?